packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -99,10 +99,12 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
       value = String(Math.max(newProps.min as number, Math.min(newProps.max as number, Number(newProps.defaultValue))));
     }
 
-    this.setState({
-      value: value,
-      precision: newProps.precision || this.state.precision
-    });
+    if (newProps.value !== undefined) {
```

Comment:
could you make a unit test? otherwise seems reasonable.

Reply:
👍 to a unit test please -- reach out if you need help with setup